### PR TITLE
Replace runtime.jvm.gc.time/runtime.jvm.gc.count metrics with process.runtime.jvm.gc.duration histogram

### PR DESCRIPTION
**Pull request by jack-berg**
Replaces #6362.

I've reduced the attributes to only record the gc name and the action that was taken (i.e. I've removed the gc cause). If needed we can add the cause later, but for now this should be sufficient to determine total time spent in GC, and categorize time spent as stop the world or parallel.

### CODE REVIEW

**Comment by PeterF778** [comment]
On second thought, removing the old GC metrics would open a gap that could be filled by the new JMX Metric Insight (PR #6573) with optional JVM metrics grouped into the "jvm" platform.

**Comment by jack-berg** [comment]
> I believe keeping the old GC time and count metrics (as non-monotonic counters) could be beneficial. This could be an option for those who want a quick look at these values without dealing with the complexities of histograms.

The view API allows you change the aggregation for instruments. If you don't need the histogram buckets, you ca configure the histogram to have a single bucket, which essentially transforms it into a sort of "summary" containing the sum, count, min, and max. You can look at just the sum and count to get the current gc time and count.

**Review by trask — APPROVED** [review]

File: instrumentation/runtime-metrics/library/src/test/java/io/opentelemetry/instrumentation/runtimemetrics/GarbageCollectorTest.java
Comment (mateuszrzeszutek) [inline instrumentation/runtime-metrics/library/src/test/java/io/opentelemetry/instrumentation/runtimemetrics/GarbageCollectorTest.java:48]: Nit: you could also pass the `NotificationEmitter` in `@Mock(extraInterfaces)`; no need for the `TestGcBean` then

**Review by mateuszrzeszutek — APPROVED** [review]

File: instrumentation/runtime-metrics/library/src/test/java/io/opentelemetry/instrumentation/runtimemetrics/GarbageCollectorTest.java
Comment (jack-berg) [reply instrumentation/runtime-metrics/library/src/test/java/io/opentelemetry/instrumentation/runtimemetrics/GarbageCollectorTest.java:65]: Nice! Didn't know such a thing existed.

File: instrumentation/runtime-metrics/library/src/test/java/io/opentelemetry/instrumentation/runtimemetrics/GarbageCollectorTest.java
Comment (jack-berg) [reply instrumentation/runtime-metrics/library/src/test/java/io/opentelemetry/instrumentation/runtimemetrics/GarbageCollectorTest.java:48]: Again, didn't know this existed. Much better!

**Comment by jack-berg** [comment]
What are folks' thoughts on merging this before or after [open-telemetry/opentelemetry-specification#2903](https://github.com/open-telemetry/opentelemetry-specification/pull/2903)? I'm slightly partial to waiting until the spec PR is merged to avoid churn for users.

**Review by mateuszrzeszutek — APPROVED** [review]

**Comment by jack-berg** [comment]
Spec [PR](https://github.com/open-telemetry/opentelemetry-specification/pull/2903) has been merged and this is unblocked!

**Comment by yorikya** [comment]
Hey folks, after upgrading from io.opentelemetry.instrumentation/opentelemetry-runtime-metrics "1.16.0-alpha" --> io.opentelemetry.instrumentation/opentelemetry-runtime-metrics "1.24.0-alpha" I had a few issues with this metric:
1. I don't get the `gc:ps_marksweep` tag anymore
2. The numbers I am getting look strange, before the upgrade I had: `avg:runtime.jvm.gc.time: 5.39K (gc:ps_scavenge)` and after the upgrade I am getting `process.runtime.jvm.gc.duration: 12 (gc:ps_scavenge)`. I know that the metric type has been changed from the gauge to the histogram
Screenshot 2023-04-16 at 15 36 03
 but the numbers look odd to me

**Comment by trask** [comment]
hi @yorikya! can you open a fresh issue for this?

**Comment by yorikya** [comment]
Sure, thanks you https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/8305

**Pull request merged**
